mater/lib/src/stores/filestore.rs

```diff
@@ -90,6 +99,86 @@ where
         }
     }
 
+    let index_offset = position;
```

Comment:
From my understanding, this chunk is writing the index out?

Reply:
Yeah this code is indeed where we write out the CARv2 index. We record the current file position in index_offset so that the CARv2 header can point to the start of this index. Right after that, the `writer.write_index(&index)` call serializes `MultihashIndexSorted` to the file, effectively appending the index bytes at position (the `index_offset`). Later, when we rewind and write the final headers, we embed `index_offset` so that any CARv2 reader knows where the index begins.